bcda/models/models.go

```diff
@@ -529,12 +524,13 @@ func GetBlueButtonID(bb client.APIClient, modelIdentifier, patientIdMode, reqTyp
 	var foundBlueButtonID = false
 	blueButtonID = patient.Entry[0].Resource.ID
 	for _, identifier := range patient.Entry[0].Resource.Identifier {
-		if strings.Contains(identifier.System, systemMode) {
-			hashedOrRawID := hashedIdentifier
+		if strings.Contains(identifier.System, "us-mbi") {
```

Comment:
Looks like we're only checking for the identifier, which is a bit worrisome.  We used to check for the actual hash value.
What if we go back to doing something like this:  https://github.com/CMSgov/bcda-app/blob/a98e431349ece5cb2dd95ac7c731fe5c496ea769/bcda/models/models.go#L504-L512
(but, of course, morph it slightly so we handle the various modes)

Reply:
which actual hashed value can we check for when we make a request via hashed hicn (due to suppression). We only get back the a hashed mbi and plaintext mbi, which is why the logic has that if statement. If you know a value we can check for when in hicn mode im all ears.

Reply:
Can't check for any value in HICN mode, due to this impeding change by BFD.  But we can in MBI mode, right?

Reply:
This looks good to me now.

Reply:
This looks good to me now.  Confirmed that the actual value is being checked (not just the identifier).